exporter/annotations.go

```diff
@@ -0,0 +1,114 @@
+package exporter
+
+import (
+	"fmt"
+
+	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"github.com/containerd/containerd/platforms"
+	"github.com/moby/buildkit/exporter/containerimage/exptypes"
+)
+
+type Annotations struct {
+	Index              map[string]string
+	IndexDescriptor    map[string]string
+	Manifest           map[string]string
+	ManifestDescriptor map[string]string
+}
+
+func NewAnnotations() *Annotations {
+	return &Annotations{
+		IndexDescriptor:    make(map[string]string),
+		Index:              make(map[string]string),
+		Manifest:           make(map[string]string),
+		ManifestDescriptor: make(map[string]string),
+	}
+}
+
+func (a *Annotations) Set(annotationType string, k string, v string) error {
```

Comment:
Can't we use the same `Parse()` logic in here as with the `CommonOpts`.

`ParseAnnotations(map) (*Annotations, rest map, error)`

Reply:
Note that if the annotation uses the specified prefix but is invalid, eg. doesn't specify an annotation name or invalid ref key then it is a parse error right away and not passed to `rest`.